gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -306,6 +308,31 @@ public void addTaskState(TaskState taskState) {
     this.taskStates.put(taskState.getTaskId(), taskState);
   }
 
+  public void addSkippedTaskState(TaskState taskState) {
+    this.skippedTaskStates.put(taskState.getTaskId(), taskState);
+  }
+
+  public void removeTaskState(TaskState taskState) {
+    this.taskStates.remove(taskState.getTaskId());
+    this.taskCount--;
+  }
+
+  /**
+   * Filter the task states corresponding to the skipped work units and add it to the skippedTaskStates
+   */
+
+  public void filterSkippedTaskStates() {
+    List<TaskState> taskStates = new ArrayList<>();
```

Comment:
Might be better to rename to skippedTaskStates, in the context of the following removeTaskState invocation.